### PR TITLE
[FIXES 169] Python version issue in docker build

### LINKED ISSUE #169: Erroring when running modules and autocompletion
## Bug Description
Attempting to run any module from pwncat results in python errors.
Using any tab completion in pwncat results in blank space

## pwncat version
Docker pwncat built from git repo.
```
$ pwncat --version
0.4.3
```

## Target System (aka "victim")
Any windows or linux machines

## Steps to Reproduce
1. Establish reverse shell.
2. Ctrl + D to go to pwncat local shell
3. Keying in "run enumerate." followed by tab key gives blank space for auto completion
4. Entering "run enumerate.gather types=file.suid" doesn't run the module but rather gives python errors

## Expected Behavior
Able to run pwncat module and use auto complete features

## Screenshots
After hitting tab
image
After running "run enumerate.gather types=file.suid"
image

Also if you require the full log please tell me if need.
Thanks!

### PR DESCRIPTION
**Pull request by Mitul16**
## Description of Changes

Fixes #169

## Major Changes Implemented:
- fix for python version: 3.9

## Pre-Merge Tasks
- [x] Formatted all modified files w/ `python-black`
- [x] Sorted imports for modified files w/ `isort`
- [x] Ran `flake8` on repo, and fixed any new problems w/ modified files
- [x] Ran `pytest` test cases
- [x] Added brief summary of updates to CHANGELOG (under `[Unreleased]`)

### CODE REVIEW

**Comment by Mitul16** [comment]
Why are we using __virtual environment__ inside docker builds?
Can't we simply use a __python__ image instead of installing it manually afterwards?

```dockerfile
FROM python:3.9-alpine3.14

# Setup dependencies
RUN set -eux \
	&& apk add --no-cache \
		alpine-sdk \
		libffi-dev \
		linux-headers \
		openssl-dev \
		musl-dev \
		cargo \
		libstdc++

# Copy pwncat source
COPY . /opt/pwncat

# Setup pip
RUN set -eux \
	&& python -m pip install -U pip setuptools wheel setuptools_rust

# Setup pwncat
RUN set -eux \
	&& cd /opt/pwncat \
	&& python setup.py install

# Download plugins and create a working directory
RUN set -eux \
	&& python -m pwncat --download-plugins \
	&& mkdir /work

# Set the working directory
WORKDIR /work

# Specify the entrypoint
ENTRYPOINT ["pwncat"]

```

**Comment by 3m0W33D** [comment]
Hi I have tested on my end as well this seems to work just fine too! Perhaps just the cache and redundant files removal can be added!

**Comment by calebstewart** [comment]
I think that using the python base image is the best path forward to prevent issues like this in the future and probably speed up image builds. 

Also, sorry for the incredibly late response here. I've been at Hacker Summer Camp over the past week and will hopefully get through all these issues/pull requests in the next couple days.

**Comment by Mitul16** [comment]
@calebstewart Please note that I have removed `pcat` from [README.md#Install](https://github.com/Mitul16/pwncat/blob/169-python-version-fix/README.md#install) in the commit https://github.com/calebstewart/pwncat/pull/171/commits/fdbfbcbbd3ab86fd9f6639647d9c41b46918a711

File: README.md [hunk 1]
```diff
@@ -129,7 +129,7 @@ be observed with your installed package.
 
 The setup script will install three binaries. They are all identical, but
 provide convenience aliases for pwncat. The three binaries are: `pwncat`,
-`pc` and `pcat`.
```
Comment (Mitul16) [inline README.md:132]: `pcat` was not being created by the `setup.py` script, possibly due to the framework overhaul

**Comment by calebstewart** [comment]
I'm going to close this request in favor of changing the base image to `python3.9-alpine`. I just implemented it and did some tests and it appears to be working. I'll open a pull request/merge momentarily after I do some final cleanup/testing.

**Pull request closed without merging**